addons/cargo/functions/fnc_removeCargoItem.sqf

```diff
@@ -0,0 +1,48 @@
+/*
+ * Author: 654wak654
+ * Removes a cargo item from the vehicle.
+ *
+ * Arguments:
+ * 0: Item Classname <STRING>
+ * 1: Vehicle <OBJECT>
+ * 2: Amount <NUMBER> (default: 1)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["item", vehicle] call ace_cargo_fnc_removeCargoItem
+ *
+ * Public: Yes
+ */
+#include "script_component.hpp"
+
+params ["_itemClass", "_vehicle", ["_amount", 1]];
+TRACE_3("params",_itemClass,_vehicle,_amount);
+
+private _loaded = _vehicle getVariable [QGVAR(loaded), []];
+
+private _itemsToRemove = [];
+private _addedSpace = 0;
+
+{
+    if (count _itemsToRemove == _amount) exitWith {};
+
+    if (_x == _itemClass || {typeOf _x == _itemClass}) then {
+        _itemsToRemove pushBack _forEachIndex;
+        ADD(_addedSpace,[_x] call FUNC(getSizeItem));
+
+        if (_x isEqualType objNull) then {
+            deleteVehicle _x;
+        };
+    };
+} forEach _loaded;
+
+{_loaded deleteAt _x} forEach _itemsToRemove;
```

Comment:
Every-time it deletes sometime, all the indexes will shift., e.g.
```
_loaded = ["ace_wheel", "ace_wheel"];
_itemsToRemove = [0,1];
// will only remove the first one::
_loaded deleteAt 0; // ["ace_wheel"];
_loaded deleteAt 1; // ["ace_wheel"];
```
I think this should work:
`{_loaded deleteAt (_x - _forEachIndex)} forEach _itemsToRemove;`